src/util/Fetcher.ts

```diff
@@ -30,11 +30,18 @@ export interface IFetcher {
 }
 
 export default class Fetcher implements IFetcher {
+  private fetcher: typeof _fetch;
+
+  constructor() {
+    if (typeof window !== "undefined" && typeof window.fetch === "function") {
```

Comment:
Or even above the level of `Fetcher` itself, but I take it there will not be a lot of instantiations?